### PR TITLE
Feature 168005 collaboration diagram

### PR DESCRIPTION
**Pull request by s-andrey**
TFS #168005
Merge this pullrequest after Flexberry/ember-flexberry-designer/pull/182

### CODE REVIEW

File: tests/dummy/app/controllers/collaboration-diagram-primitives-demo.js [hunk 1]
```diff
@@ -30,118 +30,118 @@ export default Ember.Controller.extend({
           model: _this.graph,
         });
 
-        let instance = new joint.shapes.flexberryUml.Instance({
+        let instance = new joint.shapes.flexberry.uml.Instance({
```
Comment (savolkov) [inline tests/dummy/app/controllers/collaboration-diagram-primitives-demo.js]: Modify examples, it should use new files from /objects/uml-primitives

File: addon/objects/uml-primitives/fd-uml-forward-nested-message.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  @class FdUmlCollNestedMsgForward
+  @extends FdUmlElement
+*/
+export default FdUmlElement.extend({
+
+  /**
+    The name of the class.
+
+    @property name
```
Comment (savolkov) [inline addon/objects/uml-primitives/fd-uml-forward-nested-message.js]: A typo, name => attrs

File: addon/objects/uml-primitives/fd-uml-forward-flat-message.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+/**
+  An object that describes a CollFlatMsgForward on the UML diagram.
+
+  @class FdUmlCollFlatMsgForward
+  @extends FdUmlElement
+*/
+export default FdUmlElement.extend({
+
+  /**
+    The name of the class.
```
Comment (savolkov) [inline addon/objects/uml-primitives/fd-uml-forward-flat-message.js]: A typo, change `name` to` attrs`

File: addon/objects/uml-primitives/fd-uml-forward-async-message.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+/**
+  An object that describes a CollAsyncMsgForward on the UML diagram.
+
+  @class FdUmlCollAsyncMsgForward
+  @extends FdUmlElement
+*/
+export default FdUmlElement.extend({
+
+  /**
+    The name of the class.
```
Comment (savolkov) [inline addon/objects/uml-primitives/fd-uml-forward-async-message.js]: A typo, change `name` to` attrs`

File: addon/objects/uml-primitives/fd-uml-backward-flat-message.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+  @class FdUmlCollFlatMsgBack
+  @extends FdUmlElement
+*/
+export default FdUmlElement.extend({
+
+  /**
+    The name of the class.
+
+    @property name
```
Comment (savolkov) [inline addon/objects/uml-primitives/fd-uml-backward-flat-message.js]: A typo, change `name` to` attrs`

File: addon/objects/uml-primitives/fd-uml-backward-async-message.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+  @class FdUmlCollAsyncMsgBack
+  @extends FdUmlElement
+*/
+export default FdUmlElement.extend({
+
+  /**
+    The name of the class.
+
+    @property name
```
Comment (savolkov) [inline addon/objects/uml-primitives/fd-uml-backward-async-message.js]: A typo, change `name` to` attrs`

File: addon/models/fd-dev-uml-cod.js [hunk 7]
```diff
@@ -1,9 +1,146 @@
...
+import FdUmlCompositionLink from '../objects/uml-primitives/fd-uml-composition-link';
+import FdUmlDesignPatternConnector from '../objects/uml-primitives/fd-uml-design-pattern-connector';
+import FdUmlDesignPattern from '../objects/uml-primitives/fd-uml-design-pattern';
+import FdUmlUsecaseActor from '../objects/uml-primitives/fd-uml-usecase-actor';
+import FdUmlCollFlatMsgForward from '../objects/uml-primitives/fd-uml-forward-flat-message';
+import FdUmlCollNestedMsgForward from '../objects/uml-primitives/fd-uml-forward-nested-message';
+import FdUmlCollAsyncMsgForward from '../objects/uml-primitives/fd-uml-forward-async-message';
+import FdUmlCollNestedMsgBack from '../objects/uml-primitives/fd-uml-backward-nested-message';
+import FdUmlCollFlatMsgBack from '../objects/uml-primitives/fd-uml-backward-flat-message';
+import FdUmlCollAsyncMsgBack from '../objects/uml-primitives/fd-uml-backward-async-message';
```
Comment (KollegOFF) [inline addon/models/fd-dev-uml-cod.js]: What a zoo of names?! Please use the following rules:
* The class name for `JointJS` matches, as far as possible, with the existing one on the server.
* The class name of the `ember` object is the name of the class for `JointJS` with the prefix `fd-uml-`.
* The filename is the name of the class of the `ember` object in the `dasherize` syntax.

**Pull request merged**